src/ini.rs

```diff
@@ -405,77 +456,98 @@ impl Ini {
     fn parse(&self, input: String) -> Result<Map<String, Map<String, Option<String>>>, String> {
         let mut map: Map<String, Map<String, Option<String>>> = Map::new();
         let mut section = self.default_section.clone();
+        let mut current_key: Option<String> = None;
+
         let caser = |val: &str| {
             if self.case_sensitive {
                 val.to_owned()
             } else {
                 val.to_lowercase()
             }
         };
-        for (num, lines) in input.lines().enumerate() {
-            let trimmed = match lines.find(|c: char| self.comment_symbols.contains(&c)) {
-                Some(idx) => lines[..idx].trim(),
-                None => lines.trim(),
+
+        for (num, raw_line) in input.lines().enumerate() {
+            let line = match raw_line.find(|c: char| self.comment_symbols.contains(&c)) {
+                Some(idx) => &raw_line[..idx],
+                None => raw_line,
             };
+
+            let trimmed = line.trim();
+
             if trimmed.is_empty() {
                 continue;
             }
-            match trimmed.find('[') {
-                Some(0) => match trimmed.rfind(']') {
-                    Some(end) => {
-                        section = caser(trimmed[1..end].trim());
-                    }
+
+            match (trimmed.find('['), trimmed.rfind(']')) {
+                (Some(0), Some(end)) => {
+                    section = caser(trimmed[1..end].trim());
+
+                    continue;
+                }
+                (Some(0), None) => {
+                    return Err(format!(
+                        "line {}: Found opening bracket for section name but no closing bracket",
+                        num
+                    ));
+                }
+                _ => {}
+            }
+
+            if line.starts_with(char::is_whitespace) && self.multiline {
+                let key = match current_key.as_ref() {
+                    Some(x) => x,
                     None => {
                         return Err(format!(
-                            "line {}: Found opening bracket for section name but no closing bracket",
-                            num
-                        ));
+                            "line {} started with identation but there is no current entry",
```

Comment:
```suggestion
                            "line {}: Started with identation but there is no current entry",
```
Just to keep in-line with other error messages.